connector/connect/client/jvm/src/main/scala/org/apache/spark/sql/KeyValueGroupedDataset.scala

```diff
@@ -460,14 +460,22 @@ private class KeyValueGroupedDatasetImpl[K, V, IK, IV](
   }
 
   override protected def aggUntyped(columns: TypedColumn[_, _]*): Dataset[_] = {
-    // TODO(SPARK-43415): For each column, apply the valueMap func first...
+    val aggCols: Seq[Column] = valueMapFunc match {
+      case Some(func) =>
+        val udf = SparkUserDefinedFunction(func, ivEncoder :: Nil, vEncoder).apply(col("*"))
+        // The mapValues fn is defined as: mapValuesUdf, [aggExprs]+,
+        // Upon receiving this fn, split it into mapValuesUdf and [aggExprs]+.
+        Seq(Column.fn("map_values", udf +: columns: _*))
```

Comment:
Any requirement for the name to be unique here? What if there's already a udf registered as `map_values` by the user?